node/node.go

```diff
@@ -935,6 +942,15 @@ func (node *AlgorandFullNode) oldKeyDeletionThread() {
 		node.mu.Lock()
 		node.accountManager.DeleteOldKeys(latestHdr, ccSigs, agreementProto)
 		node.mu.Unlock()
+
+		// Delete expired records from participation registry.
+		for _, record := range node.participationRegistry.GetAll() {
+			if record.LastVote < hdr.Round {
+				if err := node.participationRegistry.Delete(record.ParticipationID); err != nil {
+					node.log.Warnf("Problem deleting key from participation registry: %w", err)
+				}
+			}
+		}
```

Comment:
looks like you need to add a `node.participationRegistry.DeleteAllBeforeRound()` and pass `hdr.Round`.